src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/talismans/ResurrectedTalisman.java

```diff
@@ -0,0 +1,106 @@
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+import com.google.gson.JsonObject;
+
+import io.github.thebusybiscuit.cscorelib2.data.PersistentJsonDataType;
+import io.github.thebusybiscuit.cscorelib2.protection.ProtectableAction;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+/**
+ * This {@link SlimefunItem} is a special variant of the {@Link Talisman}. This specific {@Link Talisman} saves you
+ * from a death caused by the void and teleports to a safe location.
+ * This location is determined as follows: Talisman NBT-saved location -> Bed spawn -> World spawn
+ *
+ * @author svr333
+ *
+ */
+
+public class ResurrectedTalisman extends Talisman {
+
+    private final NamespacedKey locationKey = new NamespacedKey(SlimefunPlugin.instance(), "resurrected_location");
+
+    public ResurrectedTalisman(SlimefunItemStack item, ItemStack[] recipe) {
+        super(item, recipe, true, true, "resurrected", new PotionEffect(PotionEffectType.GLOWING, 400, 0), new PotionEffect(PotionEffectType.ABSORPTION, 400, 4), new PotionEffect(PotionEffectType.DAMAGE_RESISTANCE, 200, 20));
+
+        addItemHandler(getItemHandler());
+    }
+
+    @Nonnull
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            Location currentLoc = e.getPlayer().getLocation();
+            JsonObject json = createJsonFromLocation(currentLoc);
+            ItemStack item = e.getItem();
+
+            if (SlimefunPlugin.getProtectionManager().hasPermission(e.getPlayer(), currentLoc, ProtectableAction.PLACE_BLOCK) && item.hasItemMeta()) {
+                ItemMeta itemMeta = item.getItemMeta();
+                
+                itemMeta.getPersistentDataContainer().set(locationKey, PersistentJsonDataType.JSON_OBJECT, json);
+                item.setItemMeta(itemMeta);
+    
+                SlimefunPlugin.getLocalization().sendMessage(e.getPlayer(), "messages.talisman.resurrected-location", true);
+                return;
+            }
+
+            SlimefunPlugin.getLocalization().sendMessage(e.getPlayer(), "messages.talisman.resurrected-location-failed", true);
+        };
+    }
+
+    @Nullable
+    public Location getSavedLocation(@Nonnull ItemStack item) {
+        PersistentDataContainer pdc = item.getItemMeta().getPersistentDataContainer();
+        JsonObject json = pdc.get(locationKey, PersistentJsonDataType.JSON_OBJECT);
+
+        if (json != null) {
+            return parseLocationFromJsonObject(json);
+        } else {
+            return null;
+        }
+    }
+
+    @Nonnull
+    private JsonObject createJsonFromLocation(@Nonnull Location loc) {
```

Comment:
I would recommend just changing this. You can use BlockPosition and just store the long.